src/main/kotlin/app/hashers/CodeLongevity.kt

```diff
@@ -215,15 +218,21 @@ class CodeLongevity(private val serverRepo: Repo,
      * Scans the repo to extract code line ages.
      */
     fun scan() : CodeLineAges? {
+        var storedHead: RevCommit? = null
+        var ageData: CodeLineAges = CodeLineAges()
+
         // Load existing age data if any.
-        val iStream = try { ObjectInputStream(FileInputStream(storagePath)) }
-            catch(e : Exception ) { null }
-        val storedHead = iStream?.readObject() as RevCommit?
-        if (storedHead == head) {
-            return null
+        try {
+            val iStream = ObjectInputStream(FileInputStream(storagePath))
+            val storedHeadId = iStream.readUTF()
+            Logger.debug("Stored repo head: storedHeadId")
+            storedHead = RevWalk(repo).parseCommit(repo.resolve(storedHeadId))
+            if (storedHead == head) {
+                return null
+            }
+            ageData = (iStream.readObject() ?: CodeLineAges()) as CodeLineAges
         }
-
-        val ageData = (iStream?.readObject() ?: CodeLineAges()) as CodeLineAges
+        catch(e: FileNotFoundException) { }
```

Comment:
ObjectInputStream and FileInputStream can possibly produce a lot of different exceptions, so e: Exception is more appropriate as it generic. We should fail hashing in case of problems with loading of data for incremental updates.

Reply:
This done on purpose, if something bad or unexpected happens I want to know that. This bug was also found, because we don't hide exceptions. We may want to hide that on release builds, but that's definitely useful thing for testing and debugging.